Tests/AppTests/SearchFilterTests.swift

```diff
@@ -0,0 +1,163 @@
+            (.greaterThan, false, .greaterThan, #line),
+            (.lessThan, false, .lessThan, #line),
+            (.match, false, .equal, #line),
+            (.negativeMatch, false, .notEqual, #line),
+            
+            (.greaterThan, true, .greaterThan, #line),
+            (.lessThan, true, .lessThan, #line),
+            (.match, true, .in, #line),
+            (.negativeMatch, true, .notIn, #line),
+        ]
+        
+        matrix.forEach { comparison, isSet, sqlOperator, line in
+            XCTAssertEqual(
+                comparison.binaryOperator(isSet: isSet),
+                sqlOperator,
+                line: line
+            )
+        }
+    }
+    
+    // MARK: Filters
+    
+    func test_starsFilter() throws {
+        XCTAssertEqual(StarsSearchFilter.key, "stars")
+        XCTAssertThrowsError(try StarsSearchFilter(value: "one", comparison: .match))
+        XCTAssertEqual(try StarsSearchFilter(value: "1", comparison: .match).value, 1)
+        
+        let filter = try StarsSearchFilter(value: "1", comparison: .greaterThan)
+        let sql = renderSQL(MockPredicateBuilder.apply(filter: filter), resolveBinds: true)
+        _assertInlineSnapshot(matching: sql, as: .lines, with: """
+        "stars" > '1'
+        """)
+    }
+    
+    func test_licenseFilter() throws {
+        XCTAssertEqual(LicenseSearchFilter.key, "license")
+        XCTAssertThrowsError(try LicenseSearchFilter(value: "mit", comparison: .match))
+        XCTAssertThrowsError(try LicenseSearchFilter(value: "compatible", comparison: .greaterThan))
+        XCTAssertEqual(try LicenseSearchFilter(value: "compatible", comparison: .match).filterType, .compatible)
+        
+        let filter = try LicenseSearchFilter(value: "compatible", comparison: .match)
+        let sql = renderSQL(MockPredicateBuilder.apply(filter: filter), resolveBinds: true)
+        _assertInlineSnapshot(matching: sql, as: .lines, with: """
+        "license" IN ('afl-3.0', 'apache-2.0', 'artistic-2.0', 'bsd-2-clause', 'bsd-3-clause', 'bsd-3-clause-clear', 'bsl-1.0', 'cc', 'cc0-1.0', 'afl-3.0'0, 'afl-3.0'1, 'afl-3.0'2, 'afl-3.0'3, 'afl-3.0'4, 'afl-3.0'5, 'afl-3.0'6, 'afl-3.0'7, 'afl-3.0'8, 'afl-3.0'9, 'apache-2.0'0, 'apache-2.0'1, 'apache-2.0'2, 'apache-2.0'3, 'apache-2.0'4)
+        """)
+    }
+    
+    func test_updatedFilter() throws {
+        XCTAssertEqual(UpdatedSearchFilter.key, "updated")
+        XCTAssertThrowsError(try UpdatedSearchFilter(value: "23rd June 2021", comparison: .match))
+        XCTAssertEqual(try UpdatedSearchFilter(value: "2001-01-01", comparison: .match).date, Date(timeIntervalSinceReferenceDate: 0))
+        
+        let filter = try UpdatedSearchFilter(value: "2001-01-01", comparison: .match)
+        let sql = renderSQL(MockPredicateBuilder.apply(filter: filter), resolveBinds: true)
+        _assertInlineSnapshot(matching: sql, as: .lines, with: """
+        "last_commit_date" = '2001-01-01'
```

Comment:
I have no idea what the underlying correct SQL is for this, but I can tell you that the filter does work 😂

The test is likely not showing the true SQL here though, and that's because the `renderSQL` function just converts the bind to a string without intelligently understanding the type/context. 

Generally speaking (and I can see how this might be confusing), this test is mostly just to provide that the lhs, rhs and operator are all correct mapped (even if the SQL isn't perfect). 

Though of course I'll accept help in getting a more representative test written 👍